quinn/src/runtime.rs

```diff
@@ -6,18 +6,20 @@ use std::{
     pin::Pin,
     sync::Arc,
     task::{Context, Poll},
-    time::Instant,
 };
 
 use udp::{RecvMeta, Transmit};
 
+use crate::Instant;
+
 /// Abstracts I/O and timer operations for runtime independence
 pub trait Runtime: Send + Sync + Debug + 'static {
     /// Construct a timer that will expire at `i`
     fn new_timer(&self, i: Instant) -> Pin<Box<dyn AsyncTimer>>;
     /// Drive `future` to completion in the background
     fn spawn(&self, future: Pin<Box<dyn Future<Output = ()> + Send>>);
     /// Convert `t` into the socket type used by this runtime
+    #[cfg(feature = "net")]
```

Comment:
Hmm, I feel like trait methods that appear/disappear based on a Cargo feature aren't great. Are you still using the `Runtime` trait in your WASM setup?

Reply:
I've gone back and forth between having this gated on a feature and having this gated on the `wasm32-unknown-unknown` target.

Another option is to keep this method but then error out in all cases if invoked. This looks clean on the `quinn` side, but is error prone when e.g. some new `quinn` contribution suddenly makes use of the `wrap_udp_socket` fn outside of the `net` feature.